fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/OperatorBuilderFactory.java

```diff
@@ -16,94 +16,15 @@
-        } else if (operator instanceof LogicalHudiScanOperator) {
-            return new LogicalHudiScanOperator.Builder();
-        } else if (operator instanceof LogicalDeltaLakeScanOperator) {
-            return new LogicalDeltaLakeScanOperator.Builder();
-        } else if (operator instanceof LogicalEsScanOperator) {
-            return new LogicalEsScanOperator.Builder();
-        } else if (operator instanceof LogicalMysqlScanOperator) {
-            return new LogicalMysqlScanOperator.Builder();
-        } else if (operator instanceof LogicalSchemaScanOperator) {
-            return new LogicalSchemaScanOperator.Builder();
-        } else if (operator instanceof LogicalMetaScanOperator) {
-            return new LogicalMetaScanOperator.Builder();
-        } else if (operator instanceof LogicalJDBCScanOperator) {
-            return new LogicalJDBCScanOperator.Builder();
-        } else if (operator instanceof LogicalValuesOperator) {
-            return new LogicalValuesOperator.Builder();
-        } else if (operator instanceof LogicalTableFunctionOperator) {
-            return new LogicalTableFunctionOperator.Builder();
-        } else if (operator instanceof LogicalWindowOperator) {
-            return new LogicalWindowOperator.Builder();
-        } else if (operator instanceof LogicalUnionOperator) {
-            return new LogicalUnionOperator.Builder();
-        } else if (operator instanceof LogicalExceptOperator) {
-            return new LogicalExceptOperator.Builder();
-        } else if (operator instanceof LogicalIntersectOperator) {
-            return new LogicalIntersectOperator.Builder();
-        } else if (operator instanceof LogicalFilterOperator) {
-            return new LogicalFilterOperator.Builder();
-        } else if (operator instanceof LogicalAssertOneRowOperator) {
-            return new LogicalAssertOneRowOperator.Builder();
-        } else if (operator instanceof LogicalRepeatOperator) {
-            return new LogicalRepeatOperator.Builder();
-        } else if (operator instanceof LogicalLimitOperator) {
-            return new LogicalLimitOperator.Builder();
-        } else if (operator instanceof LogicalCTEConsumeOperator) {
-            return new LogicalCTEConsumeOperator.Builder();
-        } else if (operator instanceof LogicalCTEAnchorOperator) {
-            return new LogicalCTEAnchorOperator.Builder();
-        } else if (operator instanceof LogicalProjectOperator) {
-            return new LogicalProjectOperator.Builder();
-        } else if (operator instanceof LogicalBinlogScanOperator) {
-            return new LogicalBinlogScanOperator.Builder();
-        } else {
+    public static <T extends Operator.Builder> T build(Operator operator) {
+        String clazzName = operator.getClass().getName();
+        String builderName = clazzName + "$Builder";
+
+        try {
+            return (T) Class.forName(builderName).getConstructor().newInstance();
```

Comment:
Does this has performance problems?

Reply:
yeah, will slow X10 than if-else, but the cost is still low, about 100ms/10w call, I think the performance is OK, and you?

Reply:
BTW, we can cache all constructor by map in JVM startup

Reply:
add a cache is better